x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_charts_container.js

```diff
@@ -79,6 +89,47 @@ function ExplorerChartContainer({
   chartsService,
 }) {
   const [explorerSeriesLink, setExplorerSeriesLink] = useState('');
+  const [mapsLink, setMapsLink] = useState('');
+
+  const {
+    services: {
+      data,
+      share,
+      application: { navigateToApp },
+    },
+  } = useMlKibana();
+
+  const getMapsLink = useCallback(async () => {
+    const initialLayers = [];
+    for (const layer in ML_ANOMALY_LAYERS) {
+      if (ML_ANOMALY_LAYERS.hasOwnProperty(layer)) {
+        initialLayers.push({
+          id: htmlIdGenerator()(),
+          type: LAYER_TYPE.GEOJSON_VECTOR,
+          sourceDescriptor: AnomalySource.createDescriptor({
+            jobId: series.jobId,
+            typicalActual: ML_ANOMALY_LAYERS[layer],
+          }),
+          style: {
+            type: 'VECTOR',
+            properties: {
+              fillColor: CUSTOM_COLOR_RAMP,
+              lineColor: CUSTOM_COLOR_RAMP,
+            },
+            isTimeAware: false,
+          },
+        });
+      }
+    }
+
+    const locator = share.url.locators.get(MAPS_APP_LOCATOR);
+    const location = await locator.getLocation({
```

Comment:
Is it possible to set a query in the target Maps page, to filter the data for the partitioning field(s) in the map clicked on in the Anomaly Explorer? Rather than showing all anomalies for the job, I usually want to visualize location data just for the selected partition (`by` field).

![image](https://user-images.githubusercontent.com/7405507/160601026-94bb19b6-74ae-467e-bde5-411031244413.png)

Reply:
Added query for partition fields in 56bdb872f33bbd964a747f2b7ec923aa44e3ceca